pages/history.tsx

```diff
@@ -1,5 +1,135 @@
-export default function HistoryPage() {
+import {
+  Box,
+  Typography,
+  Button,
+  Unstable_Grid2 as Grid2,
+  useMediaQuery,
+  Autocomplete,
+  TextField,
+  Checkbox,
+  useTheme,
+} from '@mui/material'
+import SearchField from 'components/SearchField'
+import { GetServerSidePropsContext } from 'next'
+import { apiWrapper } from 'utils/apiWrappers'
+import { CategoryResponse, LogResponse } from 'utils/types'
+import { NextRouter, useRouter } from 'next/router'
+import categoriesHandler from '@api/categories'
+import { DatePicker } from '@mui/x-date-pickers'
+import HorizontalRuleIcon from '@mui/icons-material/HorizontalRule'
+import { addURLQueryParam, removeURLQueryParam } from 'utils/queryParams'
+
+export async function getServerSideProps(context: GetServerSidePropsContext) {
+  return {
+    props: { categories: await apiWrapper(categoriesHandler, context) },
+  }
+}
+
+interface HistoryPageProps {
+  logs: LogResponse[]
+  categories: CategoryResponse[]
+}
+
+const updateQuery = (router: NextRouter, key: string, val?: string) => {
```

Comment:
i've used this a lot as well. maybe we should add a tech debt to pull this into the `useQueryParams` stuff

Reply:
added a tech debt item